src/action.c

```diff
@@ -49,15 +49,19 @@
 #error SYSCONFDIR not defined
 #endif
 
+int close_fd(int fd, struct stat *st) {
+    if ( S_ISDIR(st->st_mode) || S_ISSOCK(st->st_mode) ) {
```

Comment:
Discussed on slack, let comment for others. `close_fd` return `1` to close file descriptor, `0` otherwise. Close directory and socket was done before this fix, I just took same code wrapped in `close_fd` for `src/action.c`

I have no idea why sssd open those files with each user processes. I tested on CentOS 6, first run, a socket appears in opened file descriptors, next runs socket wasn't here, probably requesting information and at same time refresh the cache ...